app.rb

```diff
@@ -0,0 +1,112 @@
+require_relative 'person'
+require_relative 'student'
+require_relative 'teacher'
+require_relative 'classroom'
+require_relative 'book'
+require_relative 'rental'
+
+class App
+  attr_reader :books, :people, :rentals, :classroom
+
+  def initialize
+    @books = []
+    @people = []
+    @rentals = []
+    @classroom = Classroom.new('college')
+  end
+
+  # Method to list all books
+  def list_all_books()
+    puts 'All Books:'
+    @books.each do |book|
+      puts "Title: #{book.title}, Author: #{book.author}"
+    end
+  end
+
+  # Method to list all people
+  def list_all_people()
+    puts 'All People:'
+    @people.each do |person|
+      puts "ID: #{person.id}, Name: #{person.name}"
+    end
+  end
+
+  # Method to create a person
+  def create_person()
+    puts 'Enter person name:'
+    name = gets.chomp
+    puts 'Enter person age:'
+    age = gets.chomp.to_i
+    puts 'Is it a student? (y/n):'
+    is_student = gets.chomp.downcase == 'y'
+
+    if is_student
+      puts 'Enter student classroom:'
+      classroom = gets.chomp
+      person = Student.new(age, name, classroom, parent_permission: true)
+    else
+      puts 'Enter teacher specialization:'
+      gets.chomp
+      person = Person.new(age, name: name, parent_permission: true)
+    end
+
+    people << person
+    puts 'Person created successfully.'
+  end
+
+  # Method to create a book
+  def create_book
+    puts 'Enter book title:'
+    title = gets.chomp
+    puts 'Enter book author:'
+    author = gets.chomp
+
+    book = Book.new(title, author)
+    @books << book
+    puts 'Book created successfully.'
+  end
+
+  # Method to create a rental
+  def create_rental
+    puts 'Enter person ID:'
+    person_id = gets.chomp.to_i
+    person = people.find { |p| p.id == person_id }
+
+    unless person
+      puts 'Person not found.'
+      return
+    end
+
+    puts 'Enter book title:'
+    book_title = gets.chomp
+    book = books.find { |b| b.title == book_title }
+
+    unless book
+      puts 'Book not found.'
+      return
+    end
+
+    puts 'Enter rental date:'
+    date = gets.chomp
+
+    rental = Rental.new(date, book, person)
+    rentals << rental
+    puts 'Rental created successfully.'
+  end
+
+  # Method to list rentals for a given person ID
+  def list_rentals_for_person()
+    puts 'Enter person ID:'
+    person_id = gets.chomp.to_i
+    person_rentals = rentals.select { |rental| rental.person.id == person_id }
+
+    if person_rentals.empty?
+      puts 'No rentals found for the given person ID.'
+    else
+      puts "Rentals for Person ID #{person_id}:"
+      person_rentals.each do |rental|
+        puts "Book: #{rental.book.title}, Date: #{rental.date}"
+      end
```

Comment:
- [ ] Good job for fixing the error, however the list of rentals is not being displayed as give in the example video.  In the current implementation, I can not see the author(s) of the rented book(s) and the details are not being displayed as seen in the video 😓 Kindly fix this implementation according to the example video 

Current UI | Expected UI 
--| ---|
![image](https://github.com/Naledi-Dikgale/oop-school-library/assets/106119105/5a01f173-fcd7-4d43-b494-dc4988a05fb0) | ![image](https://github.com/Naledi-Dikgale/oop-school-library/assets/106119105/30aabe3d-adf7-4725-b624-7e33850778b3)